### PR TITLE
Developing simple blockchain app in flutter

### LINKED ISSUE #5220: Developing simple Blockchain app In flutter
*NOTE*: All **1st-time contributors** should know Topic approval and PR submission does NOT guarantee your Topic/Article will be published. Our team of Peer Reviewers and Content Moderators will review all PRs that come in to make sure they adhere to the standard of quality we expect from the EngEd community. 

We expect all community memebers to go through provided examples, resources, and previously published material before submitting content. As a rule of thumb, please only submit articles (pull requests) that are complete, formatted correctly, and include a fully polished article (ready to be published and error free). 

All revisions and edits should be *completed on your own forked repo (as to not take up room in the queue)*. **Any PR that is submitted incomplete (meaning not ready to be published as is) will be closed.** A PR being closed does NOT mean the article can not be published (or fixed) it simply means that the edits, revisions, and fixes will happen outside of the queue.

#### Pre-submission advice
By following all our pre-submission advice and reviewing our Resources folder, you will maximise your chances of your topic being approved.

We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner.

Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published (subject to vary depending on _volume and/or backlog of articles_).

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles that will help you propose and write a successful article.

**Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

### Writing sample(s): 
Include any links or writing samples - to help our team better gauge your writing quality.

#### Developing simple Blockchain app In flutter
Developing simple Blockchain app In flutter

Use keyword research to improve your article's SEO.

#### Proposed article introduction
A blockchain is a decentralized database that everyone can access. They have an intriguing property where once data is stored in a blockchain, changing it becomes incredibly difficult.
Smart contracts (also known as decentralized applications or DApps) are similar to conventional contracts; the only distinction being they are electronic. A smart contract is a small computer program that is stored in a blockchain. They can eliminate existing third-party entities from many trustless ecosystems.
In this article we will see how one can implement the blockchain technology in our flutter applications.We are going to design a simple application that will run ion blockchain technology and will be used to keep track of transactions.
For example, if the article is based on machine learning, use the following: [Machine learning] Introduction to Machine Learning.
If the article is based on developing an Android application, use the following: [Android] Developing Apps using Android.

ONLY viable topics to pick from:
- API
- Artificial Intelligence
- Blockchain
- Containers
- Edge Computing
- Languages
- Machine Learning
- Networking
- Node.js
- Security

#### Key takeaways 
- Whats is blockchain
- what are smart contracts?
- How to build a customized blockchain for a flutter application 
- Build a simple application that allow transactions  running on a blockchain technology

#### Article quality
I have created an app called "Transactions". It shows different types of transaction happening at present time on blockchain . The transactions includes purchase, donation and sell.

I have used Material design in the app along with RecyclerView for showing information to the user . This article will be updated when new feature is added in this application.
Is yours more in-depth? Does it cover additional topics? Do you provide handy tips or anecdotal advice?

#### References
-https://hackernoon.com/learn-blockchains-by-building-one-117428612f46

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by johnniembugua**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #5220 `

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by katungi** [comment]
Hey @johnniembugua, thank you for taking your time to add to the amazing knowledge pool we are building.
 My name is Daniel, I will be taking you through the review process. Let's get started 🎊 , I am so excited to learn more as I am a flutter developer as well 🚀

**Comment by crezyj** [comment]
@katungi Thank you for your response. I am looking forward to your  guidance

**Comment by johnniembugua** [comment]
@katungi Thank you so much for the response. I am looking forward to your guidance

**Comment by Kayere** [comment]
Plagiarism check done.

File: content/articles/Developing-simple-Blockchain-app-In-flutter/index.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
+
+In this article, we will write about creating a blockchain flutter application.
+### Introduction
+ A flutter application is a  mobile app dart code that runs on iOS and Android.  Flutter uses Google's Skia library to draw widgets in UIs, which gives it high performance. Flutter also has its advanced framework that lets developers create mobile apps easily and quickly
```
Comment (katungi) [inline content/articles/Developing-simple-Blockchain-app-In-flutter/index.md:4]: The first part of that sentence needs some work. It has some grammar issues. 
Also, it's awesome that you mentioned Skia 💯

Comment (katungi) [inline content/articles/Developing-simple-Blockchain-app-In-flutter/index.md:4]: The first part of that sentence needs some work. It has some grammar issues. 
Also, it's awesome that you mentioned Skia 💯

File: content/articles/Developing-simple-Blockchain-app-In-flutter/index.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+Once you have installed the Flutter SDK, you can create a new project by running the following command in your terminal:
+```cmd
+flutter create my-blockchain-app
+```
+Now that our project is created, let's open it in your favorite editor.  I will be using Visual Studio Code for this tutorial.
+
+You should see the following code in your editor:
+
+//This is a dart file.  If you open it in another editor, make sure to associate *.dart files with dart-sdk
```
Comment (katungi) [inline content/articles/Developing-simple-Blockchain-app-In-flutter/index.md:22]: I am a bit lost by this and the following statement. Wrap that error in a code-block so it is easy to understand

**Review by katungi — COMMENTED** [review]
Hello @johnniembugua , after going through your article, I have to say that the article is really informative and well done, I learned a lot from it . However, it is riddled with minor grammatical  mistakes. Here are some suggestions to fix that:
- Go through the article from a reader's POV and see the mistakes you can fix.
- Run the article through Grammar checking software like Grammarly to help improve the quality of the article.

Once you have fixed that, please ping me so we move forward. 

**NOTE**: If you do not improve the quality of the article, It will be closed.

File: content/articles/Developing-simple-Blockchain-app-In-flutter/index.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+ body: new HomePage(), );
+       Button(
+          onClick: () {
+           network.addDevice(); 
+});
+} } 
+```
+Notice that we are connecting to my Devices, which is the name of my MQTT broker. You will need to change this to match the name of your broker. Now when you run the application on different devices, they will all be connected to the same blockchain network and will share the same blockchain data.
+### Conclusion
+Congratulations! You have now created a flutter application that uses a dart blockchain. In Part 2 of this series, we will learn how to create a proof of work algorithm and add it to our blockchain so that other devices will be able to join the network.
```
Comment (katungi) [inline content/articles/Developing-simple-Blockchain-app-In-flutter/index.md:215]: At Section we generally do not encourage series articles. However, you can suggest a more advanced application of the same topic as a new article and refer readers to this article as a prerequisite to following along.

File: content/articles/Developing-simple-Blockchain-app-In-flutter/index.md (hunk 1 above)
Comment (johnniembugua) [reply content/articles/Developing-simple-Blockchain-app-In-flutter/index.md:4]: done

**Comment by johnniembugua** [comment]
@katungi I am glad to inform you that i have mad the changes suggested.

**Comment by katungi** [comment]
Hello @johnniembugua , I have gone through the article. I really love the content. I had an easy time following it. Just a tip, it is always helpful to provide a snippet with all the code, so that the reader can finally see what the build up was leading to. It also helps the reader see exactly which piece of code goes. You can also provide a repo or remote place where the reader can access the code. 
Once you provide that, we will be ready for final review. Ping me when you are done

**Comment by johnniembugua** [comment]
@katungi done the changes requested

**Comment by katungi** [comment]
Hey, Just a quick one. You seem to have linked your personal GitHub instead of where the code is hosted. Please check on that.

**Comment by johnniembugua** [comment]
Hey @katungi , Let me rectify that.

File: content/articles/Developing-simple-Blockchain-app-In-flutter/index.md [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+
+```bash
+flutter create my-blockchain-app
+```
+
+Now that our project is created, let's open it in a code editor.  I will be using Visual Studio Code for this tutorial.
+
+When you run the application, you should see a white screen. This is because we have not added any code or widgets to our UI.
+
+### Adding widgets to our UI
```
Comment (WanjaMIKE) [inline content/articles/Developing-simple-Blockchain-app-In-flutter/index.md:39]: I do not understand how this section is related to the Blockchain app. Add and centering a `Text` widget - what is its connection.

File: content/articles/Developing-simple-Blockchain-app-In-flutter/index.md [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+Notice that we are connecting to `my Devices`, which is the name of my` MQTT` broker. You will need to change this to match the name of your broker. 
+
+When you run the application on different devices, they will all be connected to the same blockchain network and thus, share the blockchain data.
+
+### Conclusion
+Congratulations! You have now created a Flutter application that uses a Dart blockchain.
+
+This is just a simple blockchain application consisting of several devices. Each transaction is broadcasted to all the other devices
+
+You can access the full code from [this GitHub repository](https://github.com/johnniembugua/blockchain-app).
```
Comment (WanjaMIKE) [inline content/articles/Developing-simple-Blockchain-app-In-flutter/index.md:286]: The link you pasted redirects to a copy of this article. Push the complete code for this project to your Github repo. I wish to go through it.

**Comment by WanjaMIKE** [comment]
@johnniembugua Apart from the above issues, your hero image is too large. Adjust it to the required size  cc @katungi

File: content/articles/Developing-simple-Blockchain-app-In-flutter/index.md (hunk 4 above)
Comment (johnniembugua) [reply content/articles/Developing-simple-Blockchain-app-In-flutter/index.md:39]: for testing purposes

File: content/articles/Developing-simple-Blockchain-app-In-flutter/index.md (hunk 5 above)
Comment (johnniembugua) [reply content/articles/Developing-simple-Blockchain-app-In-flutter/index.md:286]: making changes to the repository adding the relevant files

File: content/articles/Developing-simple-Blockchain-app-In-flutter/index.md (hunk 4 above)
Comment (WanjaMIKE) [reply content/articles/Developing-simple-Blockchain-app-In-flutter/index.md:39]: Kindly, remove the code and add more blockchain-related content. The Flutter app comes with default widgets for testing if the app is working or not,

**Comment by katungi** [comment]
Thanks for the corrections @WanjaMIKE. Hello @johnniembugua any update on the changes?

**Comment by johnniembugua** [comment]
almost done working on the repository pushing the updates today

**Comment by katungi** [comment]
Hello @johnniembugua , how are the changes coming? Should we close the PR to clear up the queue??

**Pull request closed without merging**